pkg/serving/config_changes.go

```diff
@@ -164,7 +166,10 @@ func UpdateServiceAnnotations(service *servingv1.Service, toUpdate map[string]st
 // UpdateRevisionTemplateAnnotations updates annotations for the given Revision Template.
 // Also validates the autoscaling annotation values
 func UpdateRevisionTemplateAnnotations(template *servingv1.RevisionTemplateSpec, toUpdate map[string]string, toRemove []string) error {
-	if err := autoscaling.ValidateAnnotations(true, toUpdate); err != nil {
+	ctx := context.Background()
+	autoscalerConfig := servingconfig.FromContextOrDefaults(ctx).Autoscaler
+	autoscalerConfig.AllowZeroInitialScale = true
+	if err := autoscaling.ValidateAnnotations(ctx, autoscalerConfig, toUpdate); err != nil {
 		return err
```

Comment:
this allows users to specify `scale-init=0` during service create/update ; IIRC I think the final validation would kick in based on the defaults set at serving level? /cc @markusthoemmes